bd-logger/src/logger.rs

```diff
@@ -676,16 +681,24 @@ impl Logger {
   /// early in the launch cycle) and then enqueues the crash report processing
   /// request.
   pub fn process_crash_reports(&mut self, session: ReportProcessingSession) -> anyhow::Result<()> {
-    let crash_monitor = loop {
-      if let Some(monitor) = self.crash_monitor.lock().as_ref().cloned() {
-        break monitor;
-      }
-      std::thread::sleep(Duration::from_millis(10));
+    let crash_monitor = if let Some(monitor) = self.crash_monitor.lock().clone() {
+      monitor
+    } else if let Some(rx) = self.crash_monitor_rx.take() {
+      let monitor = rx.blocking_recv()?;
+      *self.crash_monitor.lock() = Some(monitor.clone());
+      monitor
+    } else {
+      anyhow::bail!("failed to store/receive crash_monitor instance");
```

Comment:
This is the sauce of the pr: used the cached monitor if there is one, otherwise get it from the builder transaction and store it, or else bail as something has gone wrong